app/scripts/modules/kubernetes/src/v2/manifest/wizard/ManifestEntry.tsx

```diff
@@ -1,36 +1,57 @@
 import * as React from 'react';
 import { FormikErrors } from 'formik';
+import { dump, load } from 'js-yaml';
 
 import { IWizardPageProps, wizardPage, Application } from '@spinnaker/core';
 
 import { YamlEditor } from 'kubernetes/v2/manifest/yaml/YamlEditor';
 import { IKubernetesManifestCommandData } from 'kubernetes/v2/manifest/manifestCommandBuilder.service';
 
-export interface IServerGroupBasicSettingsProps extends IWizardPageProps<IKubernetesManifestCommandData> {
+export interface IManifestBasicSettingsProps extends IWizardPageProps<IKubernetesManifestCommandData> {
   app: Application;
 }
 
-class ManifestEntryImpl extends React.Component<IServerGroupBasicSettingsProps> {
+export interface IManifestBasicSettingsState {
+  manifestString: string;
+}
+
+class ManifestEntryImpl extends React.Component<IManifestBasicSettingsProps, IManifestBasicSettingsState> {
   public static LABEL = 'Manifest';
 
+  constructor(props: IManifestBasicSettingsProps) {
+    super(props);
+
+    const { values } = this.props.formik;
+    const [first = null, ...rest] = values.command.manifests || [];
+    const manifest = rest && rest.length ? values.command.manifests : first;
+    try {
+      this.state = {
+        manifestString: manifest ? dump(manifest) : null,
+      };
+    } catch (e) {
+      this.state = { manifestString: null };
+    }
+  }
+
   public validate = (_values: IKubernetesManifestCommandData) => {
     return {} as FormikErrors<IKubernetesManifestCommandData>;
   };
 
-  private handleChange = (manifests: any) => {
+  private handleChange = (manifestString: string) => {
```

Comment:
One API suggestion but not a blocker at all: make the onChange signature 
```typescript
(raw: string, manifest: any, errors: []E)
```
so that the YamlEditor remains the only one responsible for `load`ing.

Reply:
yes, that's better.